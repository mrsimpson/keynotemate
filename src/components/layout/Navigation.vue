<template>
  <nav class="navigation" :class="{ 'menu-open': isMenuOpen }">
    <div class="nav-content">
      <div class="nav-brand">ðŸŽ¹ Piano Buddy</div>

      <button
        v-if="isMobile"
        class="burger-menu"
        :aria-expanded="isMenuOpen"
        aria-label="Toggle navigation menu"
        @click="toggleMenu"
      >
        <span></span>
        <span></span>
        <span></span>
      </button>

      <div
        v-show="!isMobile || isMenuOpen"
        :class="{ show: isMenuOpen }"
        class="nav-links"
      >
        <router-link to="/library" @click="closeMenu"
          >{{ t('navigation.library') }}
        </router-link>
        <router-link to="/collections" @click="closeMenu"
          >{{ t('navigation.collections') }}
        </router-link>
        <router-link to="/config" @click="closeMenu"
          >{{ t('navigation.configuration') }}
        </router-link>
        <LanguageSelector />
      </div>
    </div>
  </nav>
</template>

<script setup lang="ts">
import { ref } from 'vue';
import { useI18n } from 'vue-i18n';
import { useResponsive } from '../../composables/useResponsive';
import LanguageSelector from './LanguageSelector.vue';

const { t } = useI18n();
const { isMobile } = useResponsive();
const isMenuOpen = ref(false);

const toggleMenu = () => {
  isMenuOpen.value = !isMenuOpen.value;
};

const closeMenu = () => {
  isMenuOpen.value = false;
};
</script>

<style scoped>
.navigation {
  background: white;
  box-shadow: var(--shadow-sm);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1000;
}

.nav-content {
  max-width: 1400px;
  margin: 0 auto;
  padding: var(--spacing-md) var(--spacing-lg);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.nav-brand {
  font-size: 1.25rem;
  font-weight: 600;
  color: var(--primary);
}

.nav-links {
  display: flex;
  gap: var(--spacing-lg);
  align-items: center;
}

.nav-links a {
  text-decoration: none;
  color: var(--text);
  font-weight: 500;
  transition: color var(--transition-normal);
}

.nav-links a:hover {
  color: var(--primary);
}

.burger-menu {
  display: none;
  flex-direction: column;
  gap: 6px;
  background: none;
  border: none;
  cursor: pointer;
  padding: var(--spacing-sm);
}

.burger-menu span {
  display: block;
  width: 24px;
  height: 2px;
  background-color: var(--text);
  transition: var(--transition-normal);
}

@media (max-width: 800px) {
  .burger-menu {
    display: flex;
  }

  .nav-links {
    display: none;
    position: absolute;
    top: 100%;
    left: 0;
    right: 0;
    background: white;
    padding: var(--spacing-md);
    flex-direction: column;
    box-shadow: var(--shadow-md);
  }

  .nav-links.show {
    display: flex;
  }

  .menu-open .burger-menu span:nth-child(1) {
    transform: translateY(8px) rotate(45deg);
  }

  .menu-open .burger-menu span:nth-child(2) {
    opacity: 0;
  }

  .menu-open .burger-menu span:nth-child(3) {
    transform: translateY(-8px) rotate(-45deg);
  }
}

@media (max-width: 800px) {
  .nav-content {
    padding: var(--spacing-sm) var(--spacing-md);
  }

  .nav-links {
    position: absolute;
    top: 100%;
    left: 0;
    right: 0;
    background: white;
    padding: var(--spacing-md);
    flex-direction: column;
    box-shadow: var(--shadow-md);
    gap: var(--spacing-md);
  }

  .nav-links a {
    padding: var(--spacing-sm);
    width: 100%;
    text-align: center;
  }
}
</style>
