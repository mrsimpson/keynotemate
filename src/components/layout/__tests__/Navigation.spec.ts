import { mount } from '@vue/test-utils'
import { describe, expect, it, vi } from 'vitest'
import { createRouter, createWebHistory } from 'vue-router'
import { createI18n } from 'vue-i18n'
import Navigation from '@/components/layout/TheNavigation.vue'

describe('Navigation Component', () => {
  const router = createRouter({
    history: createWebHistory(),
    routes: [
      { path: '/library', component: { template: 'Library' } },
      { path: '/collections', component: { template: 'Collections' } },
      { path: '/config', component: { template: 'Config' } },
    ],
  })

  const i18n = createI18n({
    legacy: false,
    locale: 'en',
    messages: {
      en: {
        navigation: {
          library: 'Library',
          collections: 'Collections',
          configuration: 'Configuration',
        },
      },
    },
  })

  it('renders desktop navigation snapshot', () => {
    // Mock window width to simulate desktop
    vi.spyOn(window, 'innerWidth', 'get').mockReturnValue(1200)

    const wrapper = mount(Navigation, {
      global: {
        plugins: [router, i18n],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders mobile navigation snapshot', () => {
    // Mock window width to simulate mobile
    vi.spyOn(window, 'innerWidth', 'get').mockReturnValue(600)

    const wrapper = mount(Navigation, {
      global: {
        plugins: [router, i18n],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it.skip('renders mobile navigation with open menu snapshot', async () => {
    // Mock window width to simulate mobile
    vi.spyOn(window, 'innerWidth', 'get').mockReturnValue(600)

    const wrapper = mount(Navigation, {
      global: {
        plugins: [router, i18n],
      },
    })

    const burgerMenu = wrapper.find('.burger-menu')
    if (!burgerMenu.exists()) {
      throw new Error('Burger menu not found')
    }
    await burgerMenu.trigger('click')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
