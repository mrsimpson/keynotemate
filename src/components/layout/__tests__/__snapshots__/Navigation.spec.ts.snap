// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation Component > renders desktop navigation snapshot 1`] = `
"<nav data-v-a5dde78c="" class="navigation">
  <div data-v-a5dde78c="" class="nav-content">
    <div data-v-a5dde78c="" class="nav-brand">🎹 Piano Buddy</div>
    <!--v-if-->
    <div data-v-a5dde78c="" class="nav-links"><a data-v-a5dde78c="" href="/library" class="">Library</a><a data-v-a5dde78c="" href="/collections" class="">Collections</a><a data-v-a5dde78c="" href="/config" class="">Configuration</a>
      <div data-v-5c78f2f1="" data-v-a5dde78c="" class="language-selector"><button data-v-5c78f2f1="" class="selected-language" title="English"><span data-v-5c78f2f1="" class="flag">🇬🇧</span><span data-v-5c78f2f1="" class="chevron">▼</span></button>
        <!--v-if-->
      </div>
    </div>
  </div>
</nav>"
`;

exports[`Navigation Component > renders mobile navigation snapshot 1`] = `
"<nav data-v-a5dde78c="" class="navigation">
  <div data-v-a5dde78c="" class="nav-content">
    <div data-v-a5dde78c="" class="nav-brand">🎹 Piano Buddy</div>
    <!--v-if-->
    <div data-v-a5dde78c="" class="nav-links"><a data-v-a5dde78c="" href="/library" class="">Library</a><a data-v-a5dde78c="" href="/collections" class="">Collections</a><a data-v-a5dde78c="" href="/config" class="">Configuration</a>
      <div data-v-5c78f2f1="" data-v-a5dde78c="" class="language-selector"><button data-v-5c78f2f1="" class="selected-language" title="English"><span data-v-5c78f2f1="" class="flag">🇬🇧</span><span data-v-5c78f2f1="" class="chevron">▼</span></button>
        <!--v-if-->
      </div>
    </div>
  </div>
</nav>"
`;
